### ISSUE TITLE
SPAudioEngine Mute Behavior

### ISSUE THREAD

**Issue by ronyeh** [Gamua/Sparrow-Framework#52]
I added a note on Line 67 of this commit: https://github.com/Gamua/Sparrow-Framework/commit/bf413cfa7b21d6279aed241c2558b070dea7a066

Recently I upgraded my Tiny Piano app to use Sparrow 2.X's `SPAudioEngine`. However, behavior has changed from 1.X. I used to use `SPAudioSessionCategory_MediaPlayback`, because it would **continue the audio even if the device was muted**. This was important as sometimes people would leave their ringer on mute and then leave a 1-star review for my app, saying that it's dumb that the audio doesn't work in a piano app.

However, `SPAudioSessionCategory_MediaPlayback` no longer overrides the mute switch.

See the comment here: https://github.com/Gamua/Sparrow-Framework/blob/master/sparrow/src/Classes/SPAudioEngine.h#L41

The comment above suggests that the old functionality is desired, and that we should instead add a new enum to the SPAudioSessionCategory to allow users to choose AVAudioSessionCategoryMultiRoute.

In summary, I think the mappings should be like this:

`SPAudioSessionCategory_MediaPlayback => AVAudioSessionCategoryPlayback`
`SPAudioSessionCategory_MultiRoute => AVAudioSessionCategoryMultiRoute`

**Comment by racarone** [comment]
Ya this is definitely a mistype! Thanks for the catch!

I've updated the branch and deprecated `SPAudioSessionCategory` and added a `startWithCategory:` method to `SPAudioEngine` that uses `AVAudioSessionCategory` directly. The old version will now select the correct category now, but I think using the categories directly just makes more sense now.

**Comment by ronyeh** [comment]
Thanks Robert. I think that is a great solution!

**Comment by PrimaryFeather** [comment]
Thanks for the fix, Robert! Ron, sorry for the bad reviews — but it's great to see you're still a loyal Sparrow user! :smile: I hope that Tiny Piano and it's siblings are still working great for you!!

**Comment by ronyeh** [comment]
No worries! I found the regression before shipping the new update. :-)

I'm not actually using Sparrow 2.X yet, since Tiny Piano is 4 yrs old and I haven't bothered to upgrade. I just borrowed some stuff from the new SPAudioEngine and noticed that it wasn't working the same way.

**Comment by PrimaryFeather** [comment]
Makes sense! Never change a running system — still one of the most important truths in our field! :wink:

**Comment by ronyeh** [comment]
It's truly a testament to the robustness of your code, that it still works
years down the road!

**Comment by PrimaryFeather** [comment]
The principle behind that is simple: use as few iOS APIs as possible! :stuck_out_tongue_closed_eyes: 

All jokes aside, whenever one my apps stopped working in a new iOS version, it was because one of the UIKit APIs were changed or introduced a bug. I can't blame Apple, because there is a lot going on in the iOS ecosystem — but as an app developer, you're definitely on the safer side by *not* moving their APIs too much. :wink:
